test/registration/test_ndt.cpp

```diff
@@ -74,8 +74,8 @@ TEST (PCL, NormalDistributionsTransform)
   // Check again, for all possible caching schemes
   for (int iter = 0; iter < 4; iter++)
   {
-    bool force_cache = (bool) iter/2;
-    bool force_cache_reciprocal = (bool) iter%2;
+    bool force_cache = (iter/2)==0;
+    bool force_cache_reciprocal = (iter%2)==0;
```

Comment:
This reverses the value of `force_cache` and `force_cache_reciprocal`. Was that intended?

Reply:
Oh, you are right, I am going to change that to `!=0`.
Currently, because of missing parentheses, the bool cast has a higher precedence than the division/modulo, which means that actually not all four test cases are carried out (see here https://godbolt.org/z/EKx5x5). Maybe the tests are now failing because there is a bug in code that has not been executed before.

Reply:
I removed those changes from this PR and opened issue #4455. The tests were indeed failing because of that.